plugins/modules/subnet.py

```diff
@@ -237,6 +236,8 @@ def main():
 
     if not module.desired_absent:
         if module_params['network_type'] == 'IPv4':
+            if 'mask' not in module_params and 'cidr' not in module_params:
+                module.fail_json('When specifying IPv4 networks, either "mask" or "cidr" is required.')
```

Comment:
```suggestion
                module.fail_json(msg='When specifying IPv4 networks, either "mask" or "cidr" is required.')
```

This is not strictly needed, but I always feel better having named params to this super overloaded func :)